tensorboard/webapp/plugins/text/text_v2_plugin.py

```diff
@@ -0,0 +1,209 @@
+
+    It does this by selecting the zeroth coordinate for every dimension greater
+    than two.
+
+    Args:
+      arr: a numpy ndarray of dimension at least 2.
+
+    Returns:
+      A two-dimensional subarray from the input array.
+
+    Raises:
+      ValueError: If the argument is not a numpy ndarray, or the dimensionality
+        is too low.
+    """
+    if not isinstance(arr, np.ndarray):
+        raise ValueError("reduce_to_2d requires a numpy.ndarray")
+
+    ndims = len(arr.shape)
+    if ndims < 2:
+        raise ValueError("reduce_to_2d requires an array of dimensionality >=2")
+    # slice(None) is equivalent to `:`, so we take arr[0,0,...0,:,:]
+    slices = ([0] * (ndims - 2)) + [slice(None), slice(None)]
+    return arr[slices]
+
+
+def reduce_and_jsonify(text_ndarr):
+    """Take a numpy.ndarray containing strings, and convert it into a
+    json-compatible list, also squashing it to two dimensions if necessary.
+
+    If the ndarray contains a single scalar string, then that ndarray is
+    converted to a list. If it contains an array of strings,
+    that array is converted to a list.  If the array contains dimensionality
+    greater than 2, all but two of the dimensions are removed, and a squashed
+    boolean is set to true.  Returned is a list, the shape of the original
+    array, and a boolean indicating squashsing has occured.
+
+    Args:
+        text_arr: A numpy.ndarray containing strings.
+
+    Returns:
+        a tuple containing:
+            The JSON-compatible list
+            The shape of the array (before being squashed)
+            A boolean indicating if the array was squashed
+    """
+    shape = text_ndarr.shape
+    squashed = False
+    if not text_ndarr.shape:
+        # It is a scalar. Just make json-compatible and return
+        return text_ndarr.tolist(), shape, squashed
+    if len(shape) > 2:
+        squashed = True
+        text_ndarr = reduce_to_2d(text_ndarr)
+    return text_ndarr.tolist(), shape, squashed
+
+
+def create_event(wall_time, step, string_ndarray):
+    """Convert a text event into a JSON-compatible response with rank <= 2"""
+    formatted_string_array, shape, squashed = reduce_and_jsonify(string_ndarray)
+    return {
+        "wall_time": wall_time,
+        "step": step,
+        "string_array": formatted_string_array,
+        "shape": shape,
+        "squashed": squashed,
+    }
+
+
+class TextV2Plugin(base_plugin.TBPlugin):
+    """Angular Text Plugin For TensorBoard"""
+
+    plugin_name = "text_v2"
+
+    def __init__(self, context):
+        """Instantiates Angular TextPlugin via TensorBoard core.
+
+        Args:
+          context: A base_plugin.TBContext instance.
+        """
+        self._multiplexer = context.multiplexer
+        self._downsample_to = (context.sampling_hints or {}).get(
+            self.plugin_name, _DEFAULT_DOWNSAMPLING
+        )
+        if context.flags and context.flags.generic_data != "false":
```

Comment:
done (maybe check the isActive() function, with this logic, we just return false always (but per the comment in the function this shouldn't matter)

Reply:
Yeah, that's expected that it would always return false now.  There are a few other plugins that do this:
https://cs.opensource.google/tensorflow/tensorboard/+/master:tensorboard/plugins/pr_curve/pr_curves_plugin.py;l=170;drc=ba302fc490c014816068b7e4b3fd9d810e259cd5

Basically returning `False` here just means that the TensorBoard core logic is allowed to infer if the plugin is active or not, rather than asking the plugin directly.

Reply:
thanks for the explanation!